src/fsharp/FSharp.Core/printf.fs

```diff
@@ -768,103 +487,117 @@ module internal PrintfImpl =
                 let fmt = getFormat prec
                 if spec.IsWidthSpecified then
                     // width=val, prec=*
-                    box(fun v -> 
+                    ValueConverter.Make (fun v -> 
                         pad fmt spec.Width v)
                 else
                     // width=X, prec=*
-                    box(fun v -> 
+                    ValueConverter.Make (fun v -> 
                         basic fmt v)
 
         /// pad here is function that converts T to string with respect of justification
         /// basic - function that converts T to string without applying justification rules
         /// adaptPadded returns boxed function that has various number of arguments depending on if width flags has '*' value 
-        let inline adaptPadded (spec: FormatSpecifier) (basic: 'T -> string) (pad: int -> 'T -> string) = 
+        let adaptPadded (spec: FormatSpecifier) (basic: obj -> string) (pad: int -> obj -> string) : ValueConverter = 
             if spec.IsStarWidth then
-                    // width=*, prec=?
-                    box(fun v width -> 
-                        pad width v)
+                // width=*, prec=?
+                ValueConverter.Make (fun v width -> 
+                    pad width v)
             else
                 if spec.IsWidthSpecified then
                     // width=val, prec=*
-                    box(fun v -> 
+                    ValueConverter.Make (fun v -> 
                         pad spec.Width v)
                 else
                     // width=X, prec=*
-                    box(fun v -> 
+                    ValueConverter.Make (fun v -> 
                         basic v)
 
-        let inline withPaddingFormatted (spec: FormatSpecifier) getFormat  (defaultFormat: string) (f: string ->  'T -> string) left right =
+        let withPaddingFormatted (spec: FormatSpecifier) getFormat  (defaultFormat: string) (f: string ->  obj -> string) left right : ValueConverter =
             if not (spec.IsWidthSpecified || spec.IsPrecisionSpecified) then
-                box (f defaultFormat)
+                ValueConverter.Make (f defaultFormat)
             else
                 if isLeftJustify spec.Flags then
                     adaptPaddedFormatted spec getFormat f left
                 else
                     adaptPaddedFormatted spec getFormat f right
 
-        let inline withPadding (spec: FormatSpecifier) (f: 'T -> string) left right =
+        let withPadding (spec: FormatSpecifier) (f: obj -> string) left right : ValueConverter =
             if not spec.IsWidthSpecified then
-                box f
+                ValueConverter.Make f
             else
                 if isLeftJustify spec.Flags then
                     adaptPadded spec f left
                 else
                     adaptPadded  spec f right
 
-    let inline isNumber (x: ^T) =
```

Comment:
Why was this definition removed?

Reply:
It was an unnecesary use of SRTP that poisoned a lot of other code making it inline-generic SRTP also, and better dealt with in the new code as written here: https://github.com/dotnet/fsharp/pull/8907/files#diff-f42bdc58d3a07dc6e26affa6147f5759R778